### ISSUE TITLE
Why separate toolchain paths by a comma when JSON arrays are available?

### ISSUE THREAD

**Issue by lierdakil** [reasonml-editor/atom-ide-reason#10]
So... this isn't really a bug, more of a question because it seems weird to me that you would want to use something like this

`ide-reason.json:`

```json
{
  "toolchainPath": "/path/to/toolchain/one,/path/to/toolchain/two"
}
```

instead of more idiomatic

```json
{
  "toolchainPath": [ "/path/to/toolchain/one", "/path/to/toolchain/two" ]
}
```

The first one is shorter by a few characters, sure, but it's not obvious what that means from a glance. The second one is just *intuitive* in what it means.

The first one also has the problem with paths that contain a comma (and while I will agree it's not the best idea to use a comma in paths, it's perfectly possible), while the second one is perfectly robust for any paths you could think to throw at it.

(changing this will also have a side effect of making code in `joinEnvPath` less WAT -- it takes a moment to figure out that `.join(',').split(',')` isn't idempotent there -- also probably a reasonable idea to drop `trim`, because directories with spaces on either side are perfectly possible, if somewhat weird)

**Comment by zaaack** [comment]
Good point! You are right, I guess array is always better than string separated by commas, I didn't have a complete thought about this feature, it's added to fix an issue, and handling cross-platform path is always tedious.

**Comment by zaaack** [comment]
Just taked your advice, could you help to review this commit? Thanks!

https://github.com/reasonml-editor/atom-ide-reason/commit/49fd095ccb15281e366143822ff116ea291b25f1

**Comment by lierdakil** [comment]
> could you help to review this commit?

Sure. [Commented on the commit](https://github.com/reasonml-editor/atom-ide-reason/commit/49fd095ccb15281e366143822ff116ea291b25f1#commitcomment-27253692)

TL;DR: LGTM :+1: 

Feel free to close this.
